### ISSUE TITLE
Optimal way to code bound sidebearings? =? ()?

### ISSUE THREAD

**Issue by tphinney** [tphinney/science-gothic#20]
@IgorFreiberger  brought up some thoughts/questions about Thomas’ syntax choices for bound sidebearings.

> 1 Why do we use these empty ( ) in sidebearings?
> 2 The = is not necessary for linked sidebearings.
> = says FontLab VI to bind the sidebearing, but a linked sidebearing is already bound so the = is redundant. The same for calculated sidebearings when a link is part of the expression.

@kateliev says ( ) are totally wrong, but the equal sign is OK and may behave differently in places:

> Eq denotes space to bbox 
> So you could Link equal to the source or equal to the source measured from bbox
> But who knows i could be wrong

**Comment by kateliev** [comment]
Nothing to discuss really. Just look at the help.
https://help.fontlab.com/fontlab-vi/Editing-Glyph-Metrics/#linking-metrics

**Comment by kateliev** [comment]
Discussing optimal, I would suggest, from experience that ii is best to avoid multiplication and division as much as possible. In future it might mess your carefully linked font if you decide to apply tracking globally.

**Comment by servantb** [comment]
As far as the empty () syntax is concerned, the FL help says:

> l() refers to the LSB of the current glyph. It makes most sense if =l() is used in the RSB field.
> r() refers to the RSB of the current glyph. It makes most sense if =r() is used in the LSB field.

**Comment by kateliev** [comment]
Point taken, you are duplicating the lsb to the rsb. My advice... You should omit this in favor of hard coded links. So if /A has LSB 20 RSB 20 and you want to have /V same as /A. In your case you are making /V LSB=A and RSB=l()...but it will be clearer if you do /V LSB=A and RSB=A.

**Comment by IgorFreiberger** [comment]
I suggest not to use = except if necessary. Now it does nothing in an expression like =A. But internal standards of FontLab VI are not so solid because, in some sense, this is the version 1.0 of a completely new application. A future change can bring some problem with these unnecessary =. 

We already have a concrete example: in recipes, = is used to indicate a recipe in new syntax. As both syntaxes are valid, =A and A will produce the same auto layer. Anyway, if you select the glyph and go to Text > Copy Text as > Recipes, the string copied in completely different. And nobody knows until when the old syntax will be supported.

Regarding =l() I was fooled by I x l similarity. I thought it was an uppercase I, not a lowercase L. My bad!

**Comment by tphinney** [comment]
OK. At some point soon we will have a global change to go through and get
rid of unnecessary "=" in existing recipes.
We could do that by using VFJ, and doing a search and replace for "=l()"
and "=r()" and replacing them with their non-= versions.
This is one of a number of "global changes" we should probably try to do
all at the same time.

**Comment by kateliev** [comment]
I would suggest keeping the '=' as it does no harm, but CLEARLY denotes a bound contour. So it will be better for us, if we need to find all bound metrics in font to have a clear identifier, as nothing else in the VFJ uses '='. Thus we leave us a small backdoor, for future "Search and replace" sessions, a you once remove it... we will hardly put it back.

**Comment by tphinney** [comment]
You can constrain ANY TWO of: RSB, LSB, and advance width. There are legitimate reasons to pick advance width, but they are rare!

The main cases for constraining advance width:
- tabular figures & glyphs matching the tabular figure width (number space, etc.)
- characters with predefined widths such as en space and em space

In some fonts, en dash and em dash are reasonably made to equal their standard widths (half-em and full-em, respectively). In SG I believe I made the dashes in the Medium, Lt and Blk masters (and their Ctr companions) equal the standard width. But the Cnd and Ext masters I gave significantly different spacing.

**Comment by tphinney** [comment]
A good case for advance width: the /j and dotless j. Tied the advance width _and_ right sidebearing to the /i equivalents.

Also
> Point taken, you are duplicating the lsb to the rsb. My advice... You should omit this in favor of hard coded links. So if /A has LSB 20 RSB 20 and you want to have /V same as /A. In your case you are making /V LSB=A and RSB=l()...but it will be clearer if you do /V LSB=A and RSB=A.

Although clearer, it also means more dependencies. If we want to change the dependency from being /A to something else, twice as many things need to change if we have two references to the other glyph instead of one. That's why I did it the way I did: equally effective at rippling through changes, but easier for changing what you are dependent on.
